### ISSUE TITLE
absolute iframe z-index is wrong

### ISSUE THREAD

**Issue by paulrouget** [servo/servo#7796]
```html
<style>

  div, iframe {
    position: absolute;
    width: 200px;
    height: 200px;
  }

  iframe {
    display: block;
    top: 0px;
    left: 0px;
    border-width: 0;
    background: green;
  }

  div {
    left: 100px;
    top: 100px;
    background: red;
  }

</style>

<body>

  <iframe src='data:,foo'></iframe>
  <div> </div>

</body>
```

<img width="679" alt="screen shot 2015-09-30 at 09 22 24" src="https://cloud.githubusercontent.com/assets/373579/10186912/e9ae5a7e-6754-11e5-856b-7b6514cb90b0.png">

**Comment by waddlesplash** [comment]
Looks like the mimetypeless `data:` is being interpreted as HTML not plaintext. Is this already tracked somewhere?

**Comment by jdm** [comment]
Nope; please file it.

**Comment by mrobinson** [comment]
This could likely be caused by the fact that iframes always get layers. Currently only stacking contexts can have layers and they are always on the top of the display list. I confirmed that adding a z-index to the div (thus promoting it to a stacking context) fixes the ordering issue.

**Comment by pcwalton** [comment]
Probably has to wait on @mrobinson's work for multiple layers per stacking context then (or WebRender).

**Comment by mrobinson** [comment]
I think I'll try to fix this case first, since I think it is the most straight-forward.

**Comment by paulrouget** [comment]
> Probably has to wait on @mrobinson's work for multiple layers per stacking context then (or WebRender).

@pcwalton Is that gonna take a long time? Just to know if I should try to work around this bug or wait.

**Comment by mrobinson** [comment]
@paulrouget The work is complete. I'm just waiting on reviews.

**Comment by paulrouget** [comment]
@mrobinson any update on these PRs?

**Comment by mrobinson** [comment]
The PR is https://github.com/servo/servo/pull/7950. @pcwalton gave a little feedback, for which I'll upload a patch quite soon. Hopefully it can land today.
